ddtrace/internal/encoding.py

```diff
@@ -156,6 +157,48 @@ def _decode_id_to_hex(hex_id):
         return int(hex_id, 16)
 
 
+class CIAppEncoderV01(_EncoderBase):
+    content_type = "application/msgpack"
+    ALLOWED_METADATA_KEYS = ("language", "library_version", "runtime-id", "env")
+
+    def __init__(self, *args, metadata=None, **kwargs):
+        super(CIAppEncoderV01, self).__init__()
+        self._init_buffer()
+        self._metadata = metadata or dict()
+
+    def __len__(self):
+        return len(self.buffer)
+
+    def _init_buffer(self):
+        self.buffer = []
+
+    def put(self, spans):
+        self.buffer.append(spans)
+
+    def encode_traces(self, traces):
+        return self._build_payload(traces=traces)
+
+    def encode(self):
+        payload = self._build_payload()
+        self._init_buffer()
+        return payload
+
+    def _build_payload(self, traces=None):
+        normalized_spans = [CIAppEncoderV01._convert_span(span) for trace in (traces or self.buffer) for span in trace]
```

Comment:
```suggestion
    def _build_payload(self, traces):
        normalized_spans = [CIAppEncoderV01._convert_span(span) for trace in traces for span in trace]
```

I think I'd prefer this. Then you can just call `self._build_payload(self.buffer)` in `def encode(self):`. I think it would be a more clear interface rather than hiding that we'll _maybe_ use `self._buffer` (or `traces = []` then it'll use the buffer, which maybe we don't want?)

What do you think?